docs/testplan/MACsec-test-plan.md

```diff
@@ -0,0 +1,445 @@
+```txt
++-----------------------------------------------------------------------------------+
+|                                                                                   |
+| DUT                                                                               |
+|                                                                                   |
++-------+-------------------+-------------------+-------------------+-------------+-+
+        *                   *                   |                   |             |
+        *                   *                   |                   |             |
+        *.........          *.........          +.........          +.........    |
+        *        :          *        :          |        :          |        :    |
+        *        :          *        :          |        :          |        :    |
+  +-----+------+ :    +-----+------+ :    +-----+------+ :    +-----+------+ :    |
+  |VM0         | :    |VM1         | :    |VM2         | :    |VM3         | :    |
+  |(Controlled)| :    |(Controlled)| :    |            | :    |            | :    |
+  |ptf_nn_agent| :    |ptf_nn_agent| :    |ptf_nn_agent| :    |ptf_nn_agent| :    |
+  +------------+ :    +------------+ :    +------------+ :    +------------+ :    |
+                 :                   :                   :                   :    |
++----------------+-------------------+-------------------+-------------------+----+-+
+|                                                                                   |
+| PTF (ptf_nn_agent)                                                                |
+|                                                                                   |
++-----------------------------------------------------------------------------------+
+
+-----       normal link
+.....       injected link
+*****       protected link
+VM<->DUT    up link
+PTF<->DUT   down link
+```
+
+All VMs and PTF docker in the host need to install PTF NN agent. So, SONiC-mgmt-docker can use an unified interface, ptf_nn_client, to handle the packets sending operation in the servers and VMs.
+
+- Check PTF to VM traffic. Verify the traffic is truly protected by MACsec.
+
+    1. Send IPv4 packet
+
+    |   Field   |      Value      |
+    | :-------: | :-------------: |
+    | ether dst | DUT mac address |
+    |  ip src   |     1.2.3.4     |
+    |  ip dst   | VM ipv4 address |
+    |  ip ttl   |       64        |
+
+    2. Expected IPv4 packet
+
+    |   Field   |      Value      |
+    | :-------: | :-------------: |
+    | ether src | DUT mac address |
+    | ether dst | VM mac address  |
+    |  ip src   |     1.2.3.4     |
+    |  ip dst   | VM ipv4 address |
+    |  ip ttl   |       63        |
+
+    3. Send a set of above packet on the down link of DUT
+    4. The target VM should receive at least one expected above packet
+    5. In the injected port of PTF, we should get at least one expected packet encapsulated by MACsec
```

Comment:
Can SA counters used to verify the encrypted packet?

Reply:
I think it's not easy to directly used to verify the encrypted packet because there are many others traffic (BGP, LLDP and etc. )on the wire so that I don't know what the exact SA counters that should be.